### PR TITLE
Investigate and implement mechanism to pass large plotData from QC

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-653

#### Link to staging deployment URL 
https://ui-martinfosco-ui160-api70-pi.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/pipeline/pull/14
https://github.com/biomage-ltd/ui/pull/160

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/specs/models/plots-tables-schemas/PlotData.v1.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+title: Plot Data
+type: object
+properties:
+  plotData:
+    type: array
+required:
+  - plotData
```
Comment (ivababukova) [inline src/specs/models/plots-tables-schemas/PlotData.v1.yaml:7]: what about the plotDataKey parameter?

File: src/api/route-services/pipeline-response.js [hunk 2]
```diff
@@ -11,19 +11,19 @@ const plotsTableService = new PlotsTablesService();
 
 const plotsInTables = {
   cellSizeDistribution: [
-    'umisInCells',
-    'umisCellRank',
+    'cellSizeDistributionHistogram',
+    'cellSizeDistributionKneePlot',
```
Comment (marcellp) [inline src/api/route-services/pipeline-response.js:15]: Where did these changes come from in your branch?

File: src/api/route-services/pipeline-response.js [hunk 3]
```diff
@@ -72,7 +73,7 @@ const pipelineResponse = async (io, message) => {
       plotsTableService.updatePlotData(
         experimentId,
         plotUuid,
-        output.plotData,
+        output.plotDataKeys[plotUuid],
```
Comment (marcellp) [inline src/api/route-services/pipeline-response.js:76]: Here this says `plotDataKeys` but in other places it says `plotDataKey`. What is the difference between them?

**Review by marcellp — DISMISSED** [review]

File: src/api/route-services/pipeline-response.js (hunk 2 above)
Comment (cosa65) [reply src/api/route-services/pipeline-response.js:15]: This was the reason why the new plots generated by the pipeline didn't seem to be persisted (refreshing the UI after running the pipeline they were lost). It was one of the issues mentioned by oliver in his thread.

They were being used to get the plotUuids, but these Uuids seem to be outdated so when I replace them with the ones being used in the rest of the services the persistence works fine.

File: src/api/route-services/pipeline-response.js (hunk 3 above)
Comment (cosa65) [reply src/api/route-services/pipeline-response.js:76]: I am calling `plotsTableService.updatePlotData()` with one of the values in `plotDataKeys`, so inside `plotsTableService.updatePlotData()`  I decided to call this value `plotDataKey`

File: src/specs/models/plots-tables-schemas/PlotData.v1.yaml (hunk 1 above)
Comment (cosa65) [reply src/specs/models/plots-tables-schemas/PlotData.v1.yaml:7]: This validation is only on the plotData that is returned from the new S3 bucket, which is just the list of values used in the plots

Comment (ivababukova) [reply src/specs/models/plots-tables-schemas/PlotData.v1.yaml:7]: I see. I didn't know that we do validation also on what is returned by AWS. What about validating what is returned from the pipelines repo? Do we do validation on that?

**Review by ivababukova — APPROVED** [review]
lgtm

**Pull request merged**